governator-core/src/main/java/com/netflix/governator/event/ApplicationEventModule.java

```diff
@@ -0,0 +1,106 @@
+package com.netflix.governator.event;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.TypeLiteral;
+import com.google.inject.matcher.Matchers;
+import com.google.inject.spi.InjectionListener;
+import com.google.inject.spi.ProvisionListener;
+import com.google.inject.spi.TypeEncounter;
+import com.google.inject.spi.TypeListener;
+import com.netflix.governator.event.guava.GuavaApplicationEventModule;
+
+/**
+ * Adds support for passing {@link ApplicationEvent}s. Default (Guava-based) implementation
+ * can be found in {@link GuavaApplicationEventModule}
+ * 
+ * See {@link EventListener} and {@link ApplicationEventDispatcher} for usage. 
+ */
+public class ApplicationEventModule extends AbstractModule {
```

Comment:
Should be 'final' as module inheritance is discouraged due to loss of context if the module is installed multiple times